community/kernel/src/test/java/org/neo4j/test/mocking/Properties.java

```diff
@@ -82,4 +82,9 @@ public Object answer( InvocationOnMock invocation ) throws Throwable
     {
         return properties.keySet().iterator();
     }
+
+    public SortedMap<String, Object> getProperties()
+    {
+        return properties;
```

Comment:
Two reasons: first I want to minimize the impact of GC that these methods have. Second, if the client changes these maps nothing bad happens. So returning modifiable maps should be fine.